core/iwasm/libraries/thread-mgr/thread_manager.c

```diff
@@ -1302,6 +1317,34 @@ wasm_cluster_set_exception(WASMExecEnv *exec_env, const char *exception)
     data.skip = NULL;
     data.exception = exception;
 
+#if WASM_ENABLE_DUMP_CALL_STACK != 0
+    if (has_exception) {
+        /* Save the stack trace of the crashed thread into the cluster */
+        WASMModuleInstance *module_inst =
+            (WASMModuleInstance *)get_module_inst(exec_env);
+
+#if WASM_ENABLE_INTERP != 0
+        if (module_inst->module_type == Wasm_Module_Bytecode
+            && wasm_interp_create_call_stack(exec_env)) {
+            wasm_interp_dump_call_stack(exec_env, false, NULL, 0);
+            wasm_frame_vec_clone_internal(
+                (wasm_frame_vec_t *)module_inst->frames,
+                (wasm_frame_vec_t *)cluster->exception_frames);
+        }
+#endif
+
+#if WASM_ENABLE_AOT != 0
+        if (module_inst->module_type == Wasm_Module_AoT
+            && aot_create_call_stack(exec_env)) {
+            aot_dump_call_stack(exec_env, true, NULL, 0);
+            wasm_frame_vec_clone_internal(
+                (wasm_frame_vec_t *)module_inst->frames,
+                (wasm_frame_vec_t *)cluster->exception_frames);
```

Comment:
Invalid use of `module_inst->frames`, its type is `Vector *` and each element is `wasm_frame_t` or `WASMCApiFrame`, the element size is `sizeof(WASMCApiFrame)`. While element type of wasm_frame_vect_t is `wasm_frame_t *` (see WASM_DECLARE_VEC(frame, *) in wasm_c_api.h)., the element size is `sizeof(void *)`.

I guess the issue that using WASM_ENABLE_THREAD_MGR to control accessing the frame element is caused by this. The `cluster->exception_frames` should be declared as `Vector *`, and the clone algorithm need to updated accordingly.

Reply:
Yes, thanks, that's exactly what was happening there